admin/class-import.php

```diff
@@ -181,11 +181,37 @@ private function clean_up() {
 		if ( file_exists( $this->filename ) && is_writable( $this->filename ) ) {
 			unlink( $this->filename );
 		}
-		if ( file_exists( $this->file['file'] ) && is_writable( $this->file['file'] ) ) {
+		if ( ! empty( $this->file['file'] ) && file_exists( $this->file['file'] ) && is_writable( $this->file['file'] ) ) {
 			unlink( $this->file['file'] );
 		}
 		if ( file_exists( $this->path ) && is_writable( $this->path ) ) {
-			rmdir( $this->path );
+			$this->remove_directory( $this->path );
 		}
 	}
+
+	/**
+	 * Removes a directory with its content.
+	 *
+	 * @param string $directory The directory to remove.
+	 */
+	private function remove_directory( $directory ) {
```

Comment:
It might be possible the function `exec` is disabled. `safe_mode` will disable this for example.